### PR TITLE
Net app test server app changes

### PR DESCRIPTION
**Pull request by mgcouchbase**

### CODE REVIEW

File: CBLClient/Apps/CBLTestServer-Dotnet/TestServer/P2PMethods.cs [hunk 1]
```diff
@@ -35,16 +42,106 @@ public class P2PMethods
...
+
+                var rootCert = new X509Certificate2(caData);
+                var auth = new ListenerCertificateAuthenticator(new X509Certificate2Collection(rootCert));
+                urlEndpointListenerConfig.Authenticator = auth;
+            }
+
+            URLEndpointListener urlEndpointListener = new URLEndpointListener(urlEndpointListenerConfig);
+            urlEndpointListener.Start();
+            AddStatus("Started the url listener and waiting for connection..On the below port");
+            Console.WriteLine(urlEndpointListener.Port);
```
Comment (euniceatcouchbase) [inline CBLClient/Apps/CBLTestServer-Dotnet/TestServer/P2PMethods.cs:110]: there are several Console.WriteLine() calls been used, are you able to use logger in TestServer?

**Review by sridevi-15 — APPROVED** [review]
Can you make sure .Net TestServer app is successfully built with new changes ? you can put the link in the PR

File: CBLClient/Apps/CBLTestServer-Dotnet/TestServer/ListenerAuthenticatorMethods.cs [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+namespace Couchbase.Lite.Testing
+{
+    public class ListenerAuthenticatorMethods
+    {
+        public static void Create([NotNull] NameValueCollection args,
+                           [NotNull] IReadOnlyDictionary<string, object> postBody,
+                           [NotNull] HttpListenerResponse response)
+        {
+            var username = postBody["username"].ToString();
+            var password = postBody["password"].ToString();
```
Comment (euniceatcouchbase) [inline CBLClient/Apps/CBLTestServer-Dotnet/TestServer/ListenerAuthenticatorMethods.cs:20]: in a real app, password will not be sent as a plain text as a string over the wire. but I guess it should be fine with our test app. does ListenerPasswordAuthenticator takes encrypted value in its constructor?

**Review by euniceatcouchbase — APPROVED** [review]
The changes look fine, have a couple of comments, but not blocking approval. :) 
Should those .DS_store files be ignored by git?

**Pull request merged**